blog-post/blog-post.md

```diff
@@ -0,0 +1,634 @@
+to describe in future posts.
+
+## Compatibility and Interoperability
+
+Introducing such a major revision to any programming language is challenging. While a certain degree
+of breakage is inevitable (and even desired), we want to make the transition as smooth as possible
+and avoid a split in the language.
+
+We intend to keep the syntax breakage as small as possible, and expect to make significant
+alterations to the current prototype to bring syntax much more in line with Classic Solidity.
+
+As with previous breaking upgrades to Solidity, ABI compatibility will be maintained between
+versions, allowing individual contracts written in incompatible versions to interoperate and live
+side by side in the same project (this strategy is also used by Rust with their "Editions" feature).
+Thanks to our history of releasing breaking changes, much of the ecosystem tooling already supports
+multiple incompatible language versions in the same project.
+
+(TODO: check and confirm the state of things in foundry / hardhat).
+
+We are also investigating the feasibility of deeper interoperability beyond just the contract ABI.
+We expect that it will be possible to share at least free functions and interface definitions
+between language versions. We will write more concretely on this topic when we have detailed designs
+or working prototypes to share.
+
+Avoiding a python2 -> python3 style split is top of mind, and we believe that upgrades should be
+manageable and that it will be possible to carry them out in an incremental manner
+
+### Syntax Breakage
+
+The syntax you see in the examples here contains quite some differences from Classic Solidity. As
+already mentioned, we expect to bring most of the Core syntax inline with the existing Classic
+syntax. After quite some internal discussion and considering user feedback, the current feeling
+within the team is that we would prefer to prioritize smoothing the transition over introducing
+(subjectively) cleaner syntax.
+
+With that said, some degree of breakage may still be required. Right now we are considering the
+following changes:
+
+- Postfix types: Supporting the kind of rich types that Core enables with a prefix syntax is quite
+  challenging from a parser implementation point of view. Certain use cases may not be possible at
+  all, and those that are may require complex lookahead or backtracking logic. A final decision here
+  will not be made until we have completed a deeper analysis of the tradeoffs.
+- Ternary operator: Changes may be required here to resolve ambiguities in the language grammar.
+- Function type / parameter syntax: The introduction of higher order functions and the more
+  functional flavour of Core lends itself much more to the usage of function parameters. While likely not
+  strictly necessary for parsing / grammar reasons, we suspect that a simpler syntax here will likely
+  be worth the change given the benefits to readability and usability.
```

Comment:
"parameter syntax" sounds confusing here. I think you mean functions passed as parameters to other functions but it sounds as if you were talking about the syntax for defining function parameters.